.github/workflows/build-test.yml

```diff
@@ -6,86 +6,147 @@ on:
-          mkdir "C:\Temp\iceoryx2\tests"
-          mkdir "C:\Temp\iceoryx2\shm"
-          icacls "C:\Temp" /t /c /grant Everyone:F
-
       - name: Run cargo fmt
         run: cargo fmt --all -- --check
 
       - name: Run cargo clippy
         run: cargo clippy -- -D warnings
 
-      - name: Run cargo build
-        run: cargo build --workspace --all-targets
-
-      - name: Run cargo test
-        run: cargo test --workspace --no-fail-fast
-
-  windows-min-version:
-    needs: preflight-check
-    timeout-minutes: 10
-    runs-on: windows-latest
+  x86_64:
+    needs: [preflight-check, static-code-analysis]
+    strategy:
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        toolchain: [stable, 1.73.0, beta, nightly]
+        mode: 
+          - name: "release"
+            arg: "--release"
+          - name: "debug"
+            arg: ""
+    timeout-minutes: 60
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout sources
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Setup Rust
         uses: dtolnay/rust-toolchain@v1
         with:
-          toolchain: 1.73.0
+          toolchain: ${{ matrix.toolchain }}
           components: rustfmt, clippy
 
       - name: Prepare system
-        run: |
-          mkdir "C:\Temp\iceoryx2\services"
-          mkdir "C:\Temp\iceoryx2\tests"
-          mkdir "C:\Temp\iceoryx2\shm"
-          icacls "C:\Temp" /t /c /grant Everyone:F
+        run: ${{ matrix.os == 'windows-latest' && 'internal\scripts\ci_prepare_windows.bat' || ( matrix.os == 'ubuntu-latest' && './internal/scripts/ci_prepare_ubuntu.sh' || 'uname -a' ) }}
 
       - name: Run cargo build
-        run: cargo build --workspace --all-targets
+        run: cargo build --workspace --all-targets ${{ matrix.mode.arg }}
 
       - name: Run cargo test
-        run: cargo test --workspace --no-fail-fast
+        run: cargo test --workspace --no-fail-fast ${{ matrix.mode.arg }}
+
+  ### TODO: does not work yet reliable on the GitHub CI, seems to end up in an infinite loop
+  ###       current alternative is a cirrus.yml aarch64 target
+  # arm:
+  #   runs-on: ubuntu-latest
+  #   strategy:
+  #     matrix:
+  #       architecture: ["aarch64"] # ["aarch64", "armv7"]
+  #       toolchain: [ stable ] # [stable, 1.73.0, beta, nightly]
+  #       mode: ["--release", ""]
+  #   timeout-minutes: 30
+  #   steps:
+  #     - uses: actions/checkout@v4
+  #     - uses: uraimo/run-on-arch-action@v2
+  #       name: Run commands
+  #       with:
+  #         arch: ${{ matrix.architecture }}
+  #         distro: archarm_latest
+  #         run: |
+  #           ./internal/scripts/ci_prepare_archlinux.sh
+  #           rustup default ${{ matrix.toolchain }}
+  #           cargo fmt --all -- --check
+  #           cargo clippy -- -D warnings
+  #           cargo build --workspace --all-targets ${{ matrix.mode }}
+  #           cargo test --workspace --no-fail-fast ${{ matrix.mode }}
```

Comment:
Do you want to keep this and give it another shot in the future?

Reply:
Yes, at the moment the CI just hangs, but when this would work reliably I would add this.